core/src/main/java/io/grpc/internal/AutoConfiguredLoadBalancerFactory.java

```diff
@@ -247,22 +249,41 @@ PolicySelection decideLoadBalancerProvider(
         lbConfigs = ServiceConfigUtil.unwrapLoadBalancingConfigList(rawLbConfigs);
       }
       if (lbConfigs != null && !lbConfigs.isEmpty()) {
```

Comment:
lbConfig parsing wins out over haveBalancerAddress. So this order looks wrong. It looks like this code should move before the haveBalancerAddress code.